custom/enikshay/reports/views.py

```diff
@@ -39,3 +46,31 @@ def get(self, request, domain, *args, **kwargs):
                 'total': location_choice_provider.query_count(query_context.query, user)
             }
         )
+
+
+@domain_admin_required
+def duplicate_ids_report(request, domain, case_type):
+    case_type = {'voucher': CASE_TYPE_VOUCHER, 'person': CASE_TYPE_PERSON}[case_type]
+    id_property = {'voucher': VOUCHER_ID, 'person': ENIKSHAY_ID}[case_type]
+
+    accessor = CaseAccessors(domain)
+    case_ids = accessor.get_case_ids_in_domain(case_type)
```

Comment:
getting all case_ids in the domain isn't a great idea. I would suggest modifying the `SqlDomainCaseChangeProvider` (or writing something similar) to return the cases directly.

Reply:
Question about `SqlDomainCaseChangeProvider`:
```python
    def iter_all_changes(self, start_from=None):
        accessor = CaseReindexAccessor(self.domain, limit_db_aliases=self.limit_db_aliases)
        for db_alias in accessor.sql_db_aliases:
            cases = accessor.get_docs(db_alias, start_from)
            while cases:
                for case in cases:
                    yield _sql_case_to_change(case)

                start_from = case.server_modified_on
                last_id = case.id
                cases = accessor.get_docs(db_alias, start_from, last_doc_pk=last_id)
```
Why does it overwrite the `start_from` param inside the `for` loop?  Are the cases sharded by date, where all cases in a subsequent shard come after all cases in the previous shard?  If cases are assigned to shards arbitrarily, then I'd expect shards after the first to be accessed with a `start_from` equal to the most recent case modification of the previous shard.

This is also an internal-only view used for debugging, so I wasn't thinking too much about performance.  If you're concerned about the load on the server, I could also write this to instead use an ES terms aggregation with [minimum_document_count = 2](https://www.elastic.co/guide/en/elasticsearch/reference/master/search-aggregations-bucket-terms-aggregation.html#_minimum_document_count_4).  I wrote it this way to be sure of the exactness of the results, and because with the current case load it completes in just a few seconds.

Reply:
Also, I'm restricting results by case type, which looks like that function doesn't provide for.  I'd expect this to be on the order of 100k cases, do you think that's an acceptable size to load all IDs at once?

Reply:
Sorry I missed the `case_type` filter. Seems like you've thought through the scale of this so fine to go ahead as is.

Also thanks for the bug spotting: https://github.com/dimagi/commcare-hq/pull/17964